packages/core/src/codewhispererChat/editor/codelens.ts

```diff
@@ -67,6 +69,18 @@ export class TryChatCodeLensProvider implements vscode.CodeLensProvider {
         return new Promise(resolve => {
             token.onCancellationRequested(() => resolve([]))
 
+            /**
+             * Disable Codelens if
+             * - lineAnnotationController is visible (i.e not in end state)
+             * - Amazon Q chat is visible
+             */
+            const isLineAnnotationVisible = globals.context.globalState.get<string>(inlinehintKey)
+            const isAmazonQVisible = globals.context.globalState.get<boolean>(isAmazonQVisibleKey)
```

Comment:
instead of another global, this could read `ChatController.isAmazonQVisible()` .

but again, this assumes we don't already have a global for this. Please look for existing similar state. There are so many of these.